### ISSUE TITLE
Required statement management in case of complex structure (allOf ) 

### ISSUE THREAD

**Issue by LasneF** [daveshanley/vacuum#505]
given a structure as below 

```
  Cat:
      description: myNiceCat
      type: object
      required:
        - nice
      allOf:
        - $ref: '#/components/schemas/animal'
        - properties:
            nice :
              type: boolean
```

vaccum return an error  | object contains `required` fields but no `properties`  | oas-schema-check |
  providing the line containing nice
  
  
  
  tips from the boss himself @daveshanley
  
  looks it s around there 
  https://github.com/daveshanley/vacuum/blob/main/functions/openapi/schema_type.go#L261
  needs to be altered to look at polymorphic props as well as root props.

**Comment by LasneF** [comment]
here is an API spec sample that highlight the issue 

[Required.json](https://github.com/user-attachments/files/16139705/Required.json)

with the error 
53:17 | error    | object contains `required` fields but no `properties`                            | oas-schema-check      | Schemas      | $.components.schemas['Pet'].required[0]



whereas checked against  various other tool it should not raise any error

**Comment by daveshanley** [comment]
Fixed in `v0.16.9`
